### PR TITLE
feat: export `ESLint.defaultConfig`

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[x] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fix https://github.com/eslint/eslint/issues/18619

```js
const { ESLint } = require("eslint");

const configs = [ ...ESLint.defaultConfig ];
```
#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

File: tests/lib/eslint/eslint.js
Comment (amareshsm) [inline tests/lib/eslint/eslint.js:163]:
```suggestion
                assert.deepStrictEqual(ESLint.defaultConfig, defaultConfig);
```
how does spreading it affect the comparison with `defaultConfig`?

Comment (snitin315) [reply tests/lib/eslint/eslint.js:163]: No, it doesn't.

File: lib/eslint/eslint.js
Comment (mdjermanovic) [inline lib/eslint/eslint.js:532]: If we are going to freeze the default config, I think it's better to do that right away in the module where it is created rather than as a side effect of accessing it through this property. That way, we could see earlier if freezing the object breaks something.

Comment (snitin315) [reply lib/eslint/eslint.js]: Makes sense, updated 👍🏻

File: lib/types/index.d.ts
Comment (amareshsm) [inline lib/types/index.d.ts:1400]:
```suggestion
     * The default configuration that ESLint uses internally. This can change between versions, so relying on this object is not recommended.
```

File: docs/src/integrate/nodejs-api.md
Comment (amareshsm) [inline docs/src/integrate/nodejs-api.md:339]:
```suggestion
The default configuration that ESLint uses internally. This can change between versions, so relying on this object is not recommended.
```

Comment (snitin315) [reply docs/src/integrate/nodejs-api.md]: Updated 👍🏻

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Since this is a feature, it needs a second approval before merging.

**Review by JoshuaKGoldberg — APPROVED** [review]
🆒

**Review by amareshsm — APPROVED** [review]
LGTM

**Pull request merged**
